pytorch_lightning/core/lightning.py

```diff
@@ -620,9 +620,9 @@ def training_step(self, *args, **kwargs) -> STEP_OUTPUT:
         Args:
             batch (:class:`~torch.Tensor` | (:class:`~torch.Tensor`, ...) | [:class:`~torch.Tensor`, ...]):
                 The output of your :class:`~torch.utils.data.DataLoader`. A tensor, tuple or list.
-            batch_idx (int): Integer displaying index of this batch
-            optimizer_idx (int): When using multiple optimizers, this argument will also be present.
-            hiddens(:class:`~torch.Tensor`): Passed in if
+            batch_idx (``int``): Integer displaying index of this batch
+            optimizer_idx (``int``): When using multiple optimizers, this argument will also be present.
+            hiddens (``Any``): Passed in if
```

Comment:
I think basically the user can return anything here and it would be passed back to the step in the next iteration. It could also be a custom object or multiple tensors if you have multiple rnn cells.

Reply:
Correct. We don't have any constraints over what it is, as we use `apply_to_collection` to detach it.

Reply:
Could we then type it with `Collection`?

Reply:
`apply_to_collection` works for `Any`, returns the input if it's not a collection